### DISCUSSION TITLE
Arbitrary groups

### DISCUSSION THREAD

**Discussion by Viraj-10** [nativewind/nativewind#749] [Ideas]
Hey, 

Currently group feature is only supported with `active`, `hover` and `focus` selector.  
Need support for [Arbitrary groups like tailwind](https://tailwindcss.com/docs/hover-focus-and-other-states#arbitrary-groups).

## Use Case:- 

```jsx
<Pressable className={"group/checkbox p-4 bg-fuchsia-700 checked"}>
  <Text className="group-[.checked]/checkbox:text-fuchsia-100">Check Text</Text>
</Pressable>
```

### Web

<img width="305" alt="Screenshot 2024-01-18 at 12 20 35 PM" src="https://github.com/marklawlor/nativewind/assets/66306233/e75f4bb3-a0a2-4f18-9832-b3627722edbf">

### Native

<img width="364" alt="Screenshot 2024-01-18 at 12 20 53 PM" src="https://github.com/marklawlor/nativewind/assets/66306233/03fbc640-f2a9-4880-99d9-0058827566d5">

PS:- I will be happy to raise a PR for this feature. @marklawlor

**Comment by marklawlor** [comment]
This probably won't happen until the library has matured a bit. There's still a couple of unanswered questions about its implementation. Hopefully I can explain this properly :thinking:  

```
<ImageBackground className="" imageClassName="group/background bg-blend-multiply">
  <Text className="group-[.bg-blend-multiply]/background:text-fuchsia-100">Check Text</Text>
</Pressable>
```

`ImageBackground` has 2 `className` props. Does `Text` see both? Do selectors "merge" the two props together, or does it simply care that one of the props matches the selector?

What if we had this code `<ImageBackground className="group/background" imageClassName=" bg-blend-multiply">`?

I'm avoiding selectors because I don't want to make a decision on this without thinking it through. I'm allowing pseudo-class selectors because they target the component's state which is shared between all props.

Before I tackle these ideas, I first need to finalise how CSS variables work across multiple props (do inline variables in one prop affect other prop, how to children inherit from multiple props?), and from there selectors will follow a similar logic.

**Reply by Viraj-10** [reply]
ohh, I was unaware of that there can multiple `className` props. 
no worries but it's good to know it's in consideration. I will try to help with other problems. 
Great Work !!
